core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EmptyPatternOptimizer.java

```diff
@@ -58,7 +70,8 @@ public void meet(StatementPattern node)
 		Value obj = node.getObjectVar().getValue();
 		Resource[] ctx = getContexts(node.getContextVar());
 		for (RepositoryConnection member : members) {
-			if (member.hasStatement(subj, pred, obj, true, ctx)) {
+			RepositoryBloomFilter bloomFilter = bloomFilters.apply(member.getRepository());
+			if (bloomFilter == null || bloomFilter.mayHaveStatement(member, subj, pred, obj, ctx)) {
```

Comment:
It is not needed for all. If I have a memory store and a MyDifferentStore, then only MyDifferentStore would need the bloom filter as the characteristics of memorystore.hasStatement() are generally good enough.